### ISSUE TITLE
crash in Point_set_2::dfs due to stack overflow

### ISSUE THREAD

**Issue by mattanf2** [CGAL/cgal#2410]
## Issue Details

Gdal crashes while calling (probably) `CGAL::Point_set_2<SurfaceTriangulationTraits>::range_search()`.
Actual crash is inside the dfs function in `Point_set_2.h` file. It is caused because of a stack overflow (The function calls itself too many times)

## Source Code

I don't know how to recreate the problem without a lot of code but fixing it seems simple enough. remove the recursion. you can use the following code:

``` c++
void dfs(Vertex_handle v, const Circle& C, std::list<Vertex_handle>& L)
{
	struct DfsStackItem {
		Vertex_circulator vc;
		Vertex_circulator start;
	}; 
	
	Vertex_handle act = v;
	std::list<DfsStackItem> stack;
	bool pushToStack = true;
	do
	{
		DfsStackItem* top = nullptr;
		bool hadPush = pushToStack;
		if (pushToStack)
		{
			stack.push_back(DfsStackItem());
			top = &(stack.back());
			mark_vertex(act);
			top->vc = incident_vertices(act);
			top->start = top->vc;
			pushToStack = false;
		}
		else
		{
			top = &(stack.back());
			++(top->vc);
		}
		if ((hadPush) || (top->vc != top->start))
		{
			act = top->vc;
			if (!is_infinite(act)) {
				if (!is_marked(act) && !(tr_circleptori(C, act->point()) == ON_UNBOUNDED_SIDE))
					pushToStack = true;
			}
		}
		else stack.pop_back();
	
	} while (!stack.empty());
}
```

## Environment

* Operating system Windows
* Compiler: Visual studio 2015
* Release or debug mode: Release
* Specific flags used (if any):
* CGAL version: produced on 4.0.2. However the bug seams relevant for the current 4.0.10 release.

**Comment by sgiraudot** [comment]
Thanks for the report, recursion should indeed be avoided. I'll fix it.

**Comment by lrineau** [comment]
Should be fixed by #2427 in the release branches for 4.10.2 and 4.11.

**Comment by mattanf2** [comment]
I forgot the most important line in the whole function. The function doesn't do anything right now.
inside the section of if (pushToStack) add the following line

		if (pushToStack)
		{
			stack.push_back(DfsStackItem());
			top = &(stack.back());
			mark_vertex(act);
			top->vc = incident_vertices(act);
			top->start = top->vc;
			pushToStack = false;

			L.push_back(v); //<-- Add this
		}
Sorry
